src/weekdays.rs

```diff
@@ -0,0 +1,723 @@
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::Weekdays;
+    /// assert_eq!(Weekdays::MON.union(Weekdays::MON), Weekdays::MON);
+    /// assert_eq!(Weekdays::MON.union(Weekdays::TUE), Weekdays::MON | Weekdays::TUE);
+    /// assert_eq!(Weekdays::ALL.union(Weekdays::MON), Weekdays::ALL);
+    /// assert_eq!(Weekdays::ALL.union(Weekdays::EMPTY), Weekdays::ALL);
+    /// ```
+    pub const fn union(self, other: Self) -> Self {
+        Self(self.0 | other.0)
+    }
+
+    /// Returns days that are in `self` or `other` but not in both.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::Weekdays;
+    /// assert_eq!(Weekdays::MON.symmetric_difference(Weekdays::MON), Weekdays::EMPTY);
+    /// assert_eq!(Weekdays::MON.symmetric_difference(Weekdays::TUE), Weekdays::MON | Weekdays::TUE);
+    /// assert_eq!(Weekdays::ALL.symmetric_difference(Weekdays::MON), Weekdays::ALL & !Weekdays::MON);
+    /// assert_eq!(Weekdays::ALL.symmetric_difference(Weekdays::EMPTY), Weekdays::ALL);
+    /// ```
+    pub const fn symmetric_difference(self, other: Self) -> Self {
+        Self(self.0 ^ other.0)
+    }
+
+    /// Returns days that are in `self` but not in `other`.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::Weekdays;
+    /// assert_eq!(Weekdays::MON.difference(Weekdays::MON), Weekdays::EMPTY);
+    /// assert_eq!(Weekdays::MON.difference(Weekdays::TUE), Weekdays::MON);
+    /// assert_eq!(Weekdays::EMPTY.difference(Weekdays::MON), Weekdays::EMPTY);
+    /// ```
+    pub const fn difference(self, other: Self) -> Self {
+        Self(self.0 & !other.0)
+    }
+
+    /// Returns `true` if `other` contains all days in `self`.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::Weekdays;
+    /// assert!(Weekdays::MON.is_subset(Weekdays::ALL));
+    /// assert!(!Weekdays::MON.is_subset(Weekdays::EMPTY));
+    /// assert!(Weekdays::EMPTY.is_subset(Weekdays::MON));
+    /// ```
+    pub const fn is_subset(self, other: Self) -> bool {
+        self.intersection(other).0 == self.0
+    }
+
+    /// Returns `true` if `self` contains all days in `other`.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::Weekdays;
+    /// assert!(Weekdays::ALL.is_superset(Weekdays::MON));
+    /// assert!(Weekdays::MON.is_superset(Weekdays::EMPTY));
+    /// assert!(!Weekdays::MON.is_superset(Weekdays::TUE));
+    /// ```
+    pub const fn is_superset(self, other: Self) -> bool {
+        self.intersection(other).0 == other.0
+    }
+
+    /// Adds a day to the collection.
+    ///
+    /// Returns `true` if the day was new to the collection.
+    ///
+    /// # Example
+    /// ```
+    /// # use chrono::{Weekday, Weekdays};
+    /// let mut weekdays = Weekdays::MON;
+    /// assert!(weekdays.insert(Weekday::Tue));
+    /// assert!(!weekdays.insert(Weekday::Tue));
+    /// ```
+    pub fn insert(&mut self, day: Weekday) -> bool {
+        if self.contains(day) {
```

Comment:
Rewrite `insert()` and `remove()` to do an early return, dropping the `else` clause.